incubator/hnc/pkg/kubectl/tree.go

```diff
@@ -34,15 +36,24 @@ var (
 var treeCmd = &cobra.Command{
 	Use:   "tree",
 	Short: "Displays the hierarchy tree rooted at the given namespace",
-	Args:  cobra.ExactArgs(1),
+	Args:  cobra.MaximumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		nnm := args[0]
 		footnotesByMsg = map[string]int{}
 		footnotes = []string{}
-		hier := getHierarchy(nnm)
-		fmt.Println(nameAndFootnotes(hier))
-		printSubtree("", hier)
-
+		nsList := make([]string, 0)
```

Comment:
I think you could just say: `var nsList []string`, and then directly assign it below. Eg, `nsList = args` if any args are provided, or `nsList = getAllNamespaces()` if they're not. No need to append or manipulate slices since in all cases, you're just taking the entire list.